src/background/stores/assessment-card-selection-store.ts

```diff
@@ -58,6 +80,53 @@ export class AssessmentCardSelectionStore extends PersistentStore<AssessmentCard
         return defaultValue;
     }
 
+    private onAssessmentStoreChanged = async (
```

Comment:
I am not sure if this is the right approach here due to, what I imagine, will be the perf implications; converting the assessment store completely after EACH change seems quite cost-intensive.

Was the approach of adding listeners to assessment actions (and updating accordingly) not feasible?

I suspect we'll have to do a conversion like this every so often (after say load assessment) but not necessarily every change.